File: snippets/go/snippet-04.go
package main

import (
  "context"
  "fmt"

  apiclientgo "github.com/gleanwork/api-client-go"
  "github.com/gleanwork/api-client-go/models/components"
)

func searchExample(s *apiclientgo.Glean) error {
  ctx := context.Background()

  res, err := s.Client.Search.Query(ctx, components.SearchRequest{
    Query:    "quarterly business review",
    PageSize: apiclientgo.Pointer[int64](10),
  }, nil)
  if err != nil {
    return err
  }

  if res.SearchResponse == nil {
    return nil
  }

  for _, result := range res.SearchResponse.Results {
    title := ""
    if result.Title != nil {
      title = *result.Title
    }
    fmt.Printf("Title: %s\n", title)
    fmt.Printf("URL: %s\n", result.URL)
  }

  return nil
}

File: snippets/java/snippet-04.java
public class SearchService {
  private final Glean client;
  
  public SearchService(String apiToken, String serverURL) {
    this.client = Glean.builder()
        .apiToken(apiToken)
        .serverURL(serverURL)
        .build();
  }
  
  public List<SearchResult> search(String query) throws Exception {
    var response = client.client().search().query()
        .searchRequest(SearchRequest.builder()
            .query(query)
            .pageSize(10L)
            .build())
        .call();
        
    return response.searchResponse()
        .map(SearchResponse::results)
        .orElse(List.of());
  }
}
